compiler/src/dotty/tools/dotc/core/ConstraintHandling.scala

```diff
@@ -350,13 +350,24 @@ trait ConstraintHandling {
       val tpw = tp.widenSingletons
       if (tpw ne tp) && (tpw <:< bound) then tpw else tp
 
+    def widenEnum(tp: Type) =
+      val tpw = tp.widenEnumCase
+      if (tpw ne tp) && (tpw <:< bound) then tpw else tp
+
     def isSingleton(tp: Type): Boolean = tp match
       case WildcardType(optBounds) => optBounds.exists && isSingleton(optBounds.bounds.hi)
       case _ => isSubTypeWhenFrozen(tp, defn.SingletonType)
 
+    def isEnumCase(tp: Type): Boolean = tp match
```

Comment:
You might want to extract the commonalities of isEnumCase and isSingleton into a method
```scala
def widenWildcard(tp: Type) = tp match
  case WildcardType(optBounds) => optBounds
  case _ => tp
```